### PR TITLE
[ContainerRegistry] Acr set operations

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

File: sdk/containerregistry/azure-containerregistry/tests/test_container_repository_client.py [hunk 1]
```diff
@@ -117,60 +155,139 @@ def test_list_tags_descending(self, containerregistry_baseurl):
 
         assert count > 0
 
-    @pytest.mark.live_test_only
+    @pytest.mark.live_test_only  # Recordings error, recieves more than 100 headers, not that many present
```
Comment (iscai-msft) [inline sdk/containerregistry/azure-containerregistry/tests/test_container_repository_client.py:158]: are you waiting for this fix before merging?

**Review by iscai-msft — APPROVED** [review]

File: sdk/containerregistry/azure-containerregistry/tests/test_container_repository_client.py [hunk 2]
```diff
@@ -117,60 +155,139 @@ def test_list_tags_descending(self, containerregistry_baseurl):
...
+        permissions = tag_props.content_permissions
+
+        client.set_tag_properties(tag_identifier, ContentPermissions(
+            can_delete=False, can_list=False, can_read=False, can_write=False,
+        ))
+        self.sleep(10)
+
+        received = client.get_tag_properties(tag_identifier)
+
+        assert received.content_permissions.can_write == False
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/tests/test_container_repository_client.py:184]: nit: comparing boolean value with ==

**Review by kristapratico — APPROVED** [review]

File: sdk/containerregistry/azure-containerregistry/tests/test_container_repository_client.py [hunk 3]
```diff
@@ -117,60 +155,139 @@ def test_list_tags_descending(self, containerregistry_baseurl):
...
+
+        client = self.create_repository_client(containerregistry_baseurl, repository)
+
+        tag_props = client.get_tag_properties(tag_identifier)
+        permissions = tag_props.content_permissions
+
+        client.set_tag_properties(tag_identifier, ContentPermissions(
+            can_delete=False, can_list=False, can_read=False, can_write=False,
+        ))
+        self.sleep(10)
```
[the remark this answers is not in the document]
Comment (seankane-msft) [reply sdk/containerregistry/azure-containerregistry/tests/test_container_repository_client.py:180]: This is a service side issue where a GET right after a DELETE operation won't give the correct value, it takes ~5 seconds to clear the cache on the service side so we have to add delays.

File: sdk/containerregistry/azure-containerregistry/tests/test_container_repository_client.py (hunk 1 above)
Comment (seankane-msft) [reply sdk/containerregistry/azure-containerregistry/tests/test_container_repository_client.py:158]: No, I'll make the fix in the next sprint.

**Pull request merged**